core/src/main/java/it/pagopa/selfcare/mscore/core/DelegationServiceImpl.java

```diff
@@ -58,16 +59,20 @@ public Delegation createDelegation(Delegation delegation) {
     @Override
     public Delegation createDelegationFromInstitutionsTaxCode(Delegation delegation) {
 
-        List<Institution> institutionsTo = institutionService.getInstitutions(delegation.getTo(), null);
+        List<Institution> institutionsTo = institutionService.getInstitutions(delegation.getTo(), delegation.getToSubunitCode());
+        // TODO: remove filter when getInstitutions API will be fixed.
         String partnerIdentifier = institutionsTo.stream()
+                .filter(institution -> StringUtils.hasText(delegation.getToSubunitCode()) || !StringUtils.hasText(institution.getSubunitCode()))
```

Comment:
this filter could be ambiguous...you might add some comments to explain why this filter